go/common/retry/retry.go

```diff
@@ -7,12 +7,19 @@ import (
 )
 
 func Do(fn func() error, retryStrat Strategy) error {
+	return Do1(func(retry int) error {
+		return fn()
+	}, retryStrat)
+}
+
+func Do1(fn func(int) error, retryStrat Strategy) error {
```

Comment:
Shall we rename this to DoWithCount() or something so it's part of the API? I think it'd be useful quite often